### PR TITLE
More expressive titles in the change log (Lombiq Technologies: OCORE-199)

### PR DESCRIPTION
**Pull request by Piedone**
Following up from https://github.com/OrchardCMS/OrchardCore/pull/16755#discussion_r1765982886.

### CODE REVIEW

**Comment by MikeAlhayek** [comment]
I still think we should not do this. Our current format groups all the changes that are related to the same feature under one bullet point.

IMO, I think we should still continue to follow the following format:

```
# Release Version

## Breaking Changes

### Feature Name
- Change 1
- Change 2
- Change 3

## Change Logs

### Feature Name
- Change 1
- Change 2
- Change 3
```

**Comment by Piedone** [comment]
Fine by me, how about now?

File: src/docs/releases/2.1.0.md [hunk 1]
```diff
@@ -6,6 +6,8 @@ Release date: Not yet released
 
 ### External Authentication Feature
 
+#### External Authentication Logic Factored out from the Users Feature
```
Comment (MikeAlhayek) [inline src/docs/releases/2.1.0.md:9]: Maybe move the `### External Authentication Feature` to a new section **Added Features**

```suggestion
#### External authentication logic has been separated from the Users feature.
```

File: src/docs/releases/2.1.0.md [hunk 2]
```diff
@@ -32,11 +34,13 @@ Also, note the following updates in `ExternalLoginSettings`:
 - `UseScriptToSyncRoles` has been renamed to `ExternalLoginSettings.UseScriptToSyncProperties`.
 - `SyncRolesScript` has been renamed to `ExternalLoginSettings.SyncPropertiesScript`.
 
-!!! note
+!!! warning
     When updating recipes to configure `LoginSettings` or `RegistrationSettings`, ensure that the settings reflect the new class structure.
 
 ### User Registration Feature
 
+#### User Registration Feature Not Needed for External Authentication
```
Comment (MikeAlhayek) [inline src/docs/releases/2.1.0.md:42]:
```suggestion
#### **User Registration** Feature No Longer Required for External Authentication
```

File: src/docs/releases/2.1.0.md [hunk 3]
```diff
@@ -52,16 +56,22 @@ Previously, the `UsersCanRegister` property controlled which types of registrati
...
+
 In the Roles feature, there were previously `AssignRoles` and `AssignRole_{RoleName}` permissions, alongside the Users feature's `AssignRoleToUsers` and `AssignRoleToUsers_{RoleName}` permissions. As these permissions were redundant and the `AssignRoles` permissions were never actually authorized against, `AssignRoles` and `AssignRole_{RoleName}` have been removed in favor of the latter.
 
 !!! warning
     Please review all your recipes and replace occurrences of `AssignRoles` with `AssignRoleToUsers`, and `AssignRole_{RoleName}` with `AssignRoleToUsers_{RoleName}`.
 
-### New **Azure Communication Services SMS** Feature
+### Azure Communication Services SMS Feature
+
+#### New Azure Communication Services SMS Feature
```
Comment (MikeAlhayek) [inline src/docs/releases/2.1.0.md:68]: Also move this section to the bottom of the file after line 77. This way we can highlight new features away from change logs.

```suggestion

## Added Features

### Azure Communication Services SMS Feature

#### New **Azure Communication Services SMS** Feature
```

**Review by MikeAlhayek — CHANGES_REQUESTED** [review]

File: src/docs/releases/2.1.0.md (hunk 2 above)
Comment (Piedone) [reply src/docs/releases/2.1.0.md:42]: Formatting with `**` won't show a difference in how the document site is rendered. Neither should we really try to change the formatting of titles one-off.

File: src/docs/releases/2.1.0.md (hunk 1 above)
Comment (Piedone) [reply src/docs/releases/2.1.0.md:9]: This is a title, so should be capitalized like titles, and it's not a sentence thus shouldn't end with punctuation.

File: src/docs/releases/2.1.0.md (hunk 3 above)
Comment (Piedone) [reply src/docs/releases/2.1.0.md]: If we have an "Added Features" section, then having at least two levels for each feature is redundant, so I added it without it. See my note about formatting titles above.

File: src/docs/releases/2.1.0.md (hunk 1 above)
Comment (MikeAlhayek) [reply src/docs/releases/2.1.0.md:9]: fine. the suggestion was updated

File: src/docs/releases/2.1.0.md (hunk 2 above)
Comment (MikeAlhayek) [reply src/docs/releases/2.1.0.md:42]: right because it's already highlighted. updated.

**Comment by MikeAlhayek** [comment]
@Piedone I would also either move the addition `External Authentication` feature to `New Features` section, or add a reference to the new `External Authentication` feature in the `New Features` section.

**Comment by Piedone** [comment]
I added it at the bottom. However, now the original section above doesn't make sense. So, I moved that under the Users feature, which is the one that actually changed.

File: src/docs/releases/2.1.0.md [hunk 4]
```diff
@@ -52,17 +56,27 @@ Previously, the `UsersCanRegister` property controlled which types of registrati
...
+
+## Added Features
+
+### Azure Communication Services SMS Feature
+
+A new feature was added to allow you to send SMS messages using Azure Communication Services (ACS). Simply enable it then navigate to the admin dashboard > `Configurations` >> `Settings` >> `SMS` to configure the provider. For more information you can refer to the [docs](../reference/modules/Sms.Azure/README.md).
+
+### External Authentication Feature
+
+The new **External Authentication** feature contains common functionality used by multiple external (i.e. non-local) authentication providers (like Microsoft or Google login). See [above](#external-authentication-logic-has-been-separated-from-the-users-feature) for details.
```
Comment (MikeAlhayek) [inline src/docs/releases/2.1.0.md:82]:
```suggestion
The new **External Authentication** feature contains common functionality used by multiple external (i.e. non-local) authentication providers (like Microsoft or Google login). See [above](#external-authentication-logic-has-been-separated-from-the-users-feature) for details.

```

**Review by MikeAlhayek — APPROVED** [review]

**Pull request merged**